src/framework/components/model/component.js

```diff
@@ -659,7 +659,11 @@ Object.assign(pc, function () {
                 }
             }
 
-            this._model = value;
+            if (value && value.immutable) {
+                this._model = value.clone();
+            } else {
+                this._model = value;
```

Comment:
if you set the immutable flag here instead, wouldn't it also fix the case where a user assigns a non-asset model to multiple components?

Reply:
It will, but it might have larger impact on number of clones then. I would need to test it on real projects to see how it impacts them.

Reply:
Going through the code, if a user assigns the same model to two different components:
- the first component will define the model's transform (since the second call to GraphNode.addChild fails)
- the second component will define the model's castShadows/receiveShadows/isStatic/lightmapped flags
- the model will only get rendered once, because layer won't add a meshinstance twice

Given this, it's probably safe to return from this function without setting the model if the model has already been set on another component.

Reply:
If the cloning when used wont be an option to fix this big term.
Then we can do as you suggested, don't crash, but return gracefully and in #debug mode console.log about it.